pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -171,22 +171,20 @@ func TestSetPutEnv(t *testing.T) {
 				err := tx.PutEnv(env)
 				require.ErrorIs(t, err, tc.wantPutError)
 
-				if tc.wantPutError != nil || tc.value == nil {
-					environment, err := tx.GetEnvList()
-					require.NoError(t, err)
-					require.Equal(t, map[string]string{}, environment)
-				} else {
-					environment, err := tx.GetEnvList()
-					require.NoError(t, err)
-
-					if tc.wantValue != nil {
-						require.Equal(t, map[string]string{
-							tc.env: *tc.wantValue}, environment)
-					} else {
-						require.Equal(t, map[string]string{tc.env: *tc.value},
-							environment)
+				wantEnv := map[string]string{}
+				if tc.wantPutError == nil {
+					if tc.value != nil {
+						wantEnv = map[string]string{tc.env: *tc.value}
+					}
+					if tc.value != nil && tc.wantValue != nil {
+						wantEnv = map[string]string{tc.env: *tc.wantValue}
 					}
 				}
+				gotEnv, err := tx.GetEnvList()
+				require.NoError(t, err,
+					"Always put msgs on the requires so you can pinpoint the failure faster")
+				require.Equal(t, wantEnv, gotEnv,
+					"Always put msgs on the requires so you can pinpoint the failure faster")
```

Comment:
These strings were only placeholders for you to put a message in the requires. Can you add ones that make sense in this context? IMHO, it should be something like "tx.GetEnvList should not return an error, but did." and "tx.GetEnvList should have returned the expected output, but did not."

Reply:
In however general I look mostly at line errors on failures, so I think putting the description makes much sense when the requirement we do isn't something that could be straight forward understood from code.

Reply:
I agree that `require` already shows you the diff between what should be vs. what is and so on, but the message really helps to figure out exactly __which__ assertion failed. This is especially useful when dealing with functions marked as `t.Helper()`, as line information is skipped for test helper functions.
IMHO, if we can have any "help" at all for pinpointing the point of failure, we should use it.